### PR TITLE
Enabling pokemon list caching

### LINKED ISSUE #1: Connect to PokeAPI
Conneting to a Rest api in android requires a couple of steps:

1. Set up a Rest Client (i suggest Okhttp) in the app
2. Connect to the PokeApi
3. get the first 100 pokemon from the list using (https://pokeapi.co/api/v2/pokemon?limit=100&offset=0)

### LINKED ISSUE #2: Show a list of pokemon names on the main screen
Following #1, we will have a set of data which can be rendered on the screen!

A huge part of a lot of Android apps is showing a list of items on a screen. There are a bunch of things we can do with this, but this task will just keep it simple. Just show a list of the pokemon names that you get from completing #1.

- [ ] The screen should show some sort of loading state while we're waiting for the data to come back from the API.
- [ ] You should be able to scroll up and down through the names of Pokemon.
- [ ] (Extra points) Rotating the device to switch between landscape/portrait shouldn't require re-fetching the data! 

Tip: Look in to how a `RecyclerView` works, and use that as the view to render the list.

### PR DESCRIPTION
**Pull request by lahiru-b**
Enabled caching of the list of Pokemon, cache is valid for 30 minutes

closes #1
closes #2

### CODE REVIEW

**Comment by lahiru-b** [comment]
This closes Issue #1 and Issue #2

**Comment by DylanLange** [comment]
if you write it like:

closes #1 
closes #2 

I think that'll make it automatic. the wording is quite particular 😛

**Comment by DylanLange** [comment]
more info here: https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

File: app/src/main/java/com/graddex/graddex/MainActivity.kt [hunk 1]
```diff
@@ -79,20 +85,4 @@ class MainActivity : AppCompatActivity() {
         })
     }
 
-    /**
```
Comment (AniFichadia) [inline app/src/main/java/com/graddex/graddex/MainActivity.kt:82]: Good little bit of cleanup :)

**Review by DylanLange — APPROVED** [review]

**Pull request merged**